src/main/java/servlets/UploadReceiptServlet.java

```diff
@@ -60,20 +62,23 @@
   private final BlobstoreService blobstoreService;
   private final BlobInfoFactory blobInfoFactory;
   private final DatastoreService datastore;
+  private final UserService userService;
   private final Clock clock;
 
   public UploadReceiptServlet() {
     this.blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
     this.blobInfoFactory = new BlobInfoFactory();
     this.datastore = DatastoreServiceFactory.getDatastoreService();
+    this.userService = UserServiceFactory.getUserService();
     this.clock = Clock.systemDefaultZone();
   }
 
   public UploadReceiptServlet(BlobstoreService blobstoreService, BlobInfoFactory blobInfoFactory,
-      DatastoreService datastore, Clock clock) {
+      DatastoreService datastore, UserService userService, Clock clock) {
```

Comment:
Does it make a difference if the UserService is initialized in the constructor vs passed in as a parameter in the test?